### PR TITLE
Completes ncnews-8: GET /api/articles

### PR DESCRIPTION
**Pull request by garetheverson**
***ADDED***
app.get('/api/articles', getArticles)
articles.models - selectArticles
articles.controllers - getArticles
Tests for valid 200 responses where an articles have, or have not got comments, with examples

***CHANGED***
No further changes

***UPDATED***
Changed variable name 'votesIsNan' to 'votesIsAString' in test '400: Responds with 'Votes property must be a number' if inc_votes not a number', per ncnews-4 review by @giteron: 'The wording may suggest it holds either a boolean, or a literal NaN (which is a typeof number), whereas currently it holds a string'

ADDENDUM: Updated /api/topics test to use jest-sorted to verify articles are returned in descending created_at date order, see https://github.com/garetheverson/back-end-news-portfolio-project/pull/7/commits/41e2e01d725d11aca481dba836eac1811d290b03

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -19,6 +20,24 @@ exports.selectArticleById = (article_id) => {
...
 
+exports.selectArticles = () => {
+  const query = `SELECT u.username as author, a.title,a.article_id, a.topic,a.created_at, a.votes, COUNT(c.article_id) as comment_count
+  FROM articles a
+  INNER JOIN  users u
+  ON a.author = u.username
+  LEFT JOIN comments c
+  on a.article_id = c.article_id
+  GROUP BY u.username, a.title,a.article_id, a.topic,a.created_at, a.votes
+  ORDER BY a.created_at DESC;`;
```
Comment (Liam310) [inline models/articles.models.js:31]: Great job with the aggregating and the left join! Couple of pointers though:
- If you want to get all the columns from a table, a nice shorthand is `articles.*` - saves you having to type them all out
- The `author`'s name is actually accessible to us within the `articles` table, so there's no need to join to the `users` table here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -203,3 +204,47 @@ describe('Success path: GET requests for users: /api/users', () => {
...
+        });
+        expect(res.body.articles[10]).toEqual({
+          author: 'icellusedkars',
+          title: 'Am I a cat?',
+          article_id: 11,
+          topic: 'mitch',
+          created_at: '2020-01-15T22:21:00.000Z',
+          votes: 0,
+          comment_count: '0',
+        });
```
Comment (Liam310) [inline __tests__/app.test.js:247]: Not sure if these extra assertions were to be doubly sure about the sort order (or maybe something else I'm missing?) From what I can see it looks to be that they are a little bit surplus to requirements - all the ones you did above give you plenty of coverage.

**Comment by Liam310** [comment]
Looks good! Have left a couple of code comments above but otherwise looking very nice and clean 👍

File: __tests__/app.test.js (hunk 2 above)
Comment (garetheverson) [reply __tests__/app.test.js:247]: They were @Liam310 👍🏻

**Pull request merged**